### DISCUSSION TITLE
Tine Groupware Community-Onboarding

### DISCUSSION THREAD

**Discussion by lab-at-nohl** [tine-groupware/tine#19] [Show and tell]
Dear fellow Community

I hope you found well from *Tine 2.0* to *Tine Groupware* (as it is called officially now) or simply call it **Tine**. 

Probably you have questions or rather want to discuss the latest developments. Or maybe you are simply new to the project. End of 2023 the Repo for the *Community Edition* has been sent to archive, while users/admins/operators are encouraged to participate here. 

That's what I do hereby. I myself am a community user **not affiliated** to the company behind Tine, *Metaways*. However, I use Tine Groupware **for more than 10 years** (with LDAP and IMAP). See my recent experience below. 

I would like to participate in an active Community, I can help in _German and English_. So, **hello there.**  

Best regards
Johannes

+++

Recently I answered some questions for myself, please see below. Yesterday I have done the **upgrade from 2023.12.1 to 2023.11.3**. It's not obvious but you need to get your *Community* installation to the first mentioned version *) to switch to the latter version, which runs on the same code base as the *Business Edition*.  

1. **Are there any new requirements?** For me all I had to do was enabling **PHP 8.2**. At least PHP 8.0 is hardcoded and scripts die without any notice if this is not met (I learned). 
2. **Is Tine still free?** The Software did not change its license. It is open source. However, although Community and Business Edition share the same code (which is not totally new, obviously the old Repo is very similar) they target a different audience. 
3. **Do I need a license key?** You do _not currently_, which actually was a surprise for myself, too. Without key some functionalities **) are locked but only a few. Thus, Tine as it has been delivered in the "allinone"-packages before, remains free. This may change in future. However, today you should go with the flow and update.  
4. **Is ActiveSync for mobile phones still available?** Yes, it works well as you may already know. It belongs to the "free" features, no license key necessary. 
5. **Can I install Tine on my webspace? Or do I need Docker containers?** No, for now you can run Tine as you have always done. All you need is to extract the Tar-ball in the right folder and re-use your config.inc.php. Besides, also Docker containers exist. Especially in a more complex installation you may want to use them as they are preconfigured. 

+++

Footnotes

*) Updating ancient versions is highly recommended now to avoid disconnection from future developments! 

**) List of **locked apps/functionalities** (as of version)
```php
        'CashBook'                                      => '*',
        'ContractManager'                               => '*',
        'DFCom'                                         => '>=2.0',
        'EFile'                                         => '*',
        'GDPR'                                          => '>=2.0',
        'HumanResources.workingTimeAccounting'          => '>=2.0',
        'KeyManager'                                    => '*',
        'MeetingManager'                                => '*',
        'OnlyOfficeIntegrator'                          => '*',
        'Tinebase.featureCreatePreviews'                => '>=2.0',
        'UserManual'                                    => '>=2.0',
```

I could update a Community installation ("allinone"-Apps) with some users flawless.

**Comment by pschuele** [comment]
hi Johannes,
thanks for the great summary & on-boarding information :+1: 

one small addendum:
PHP 8.2 is not fully supported yet - we are working on it but in the meantime, PHP 8.0 or 8.1 should be used.

**Comment by ignoreigor** [comment]
Hi, 
nach dem Abschied der Community-Schiene, ist mir das Versionierungsschema nun gar nicht mehr klar, s. https://github.com/tine-groupware/tine/issues/28#issuecomment-2014881385 bzw. https://github.com/tine-groupware/tine/tags.

Möglicher Erklärungsversuch: Es gibt LTS-Release, aktuell 2023.11.6. Und ein aktuelles (oder ist es, weil nicht angekündigt, Developement?) Release 2024.11-2024.03.14.1? Von beiden gibt es Weeklies, die aber keine stabilen Releases darstellen. 
Einmal auf 2024.xx sollte, man nicht auf 2023.xx wechseln, auch wenn letzteres neuer wäre.

Stimmt das so ungefähr? Ansonsten würde ich mir den Podcast 16 nochmal mit mehr Hingabe anhören und es dann vielleicht verstehen.

**Reply by lab-at-nohl** [reply]
Ja, verworren ,🤔.

20xx.11.y war früher die Business Edition. 20xx.12.y die Community Edition (CE).

*CE wird nicht weitergeführt*. Deshalb gibt es mit 2023.11.3 einmalig den Weg, von einer CE 20xx.12.y - also von CE zu Business - zu upgraden (auch wenn es wie ein Downgrade aussieht). Danach ist man wieder im Entwicklungspfad.

Der Entwicklungspfad beinhaltet insbesondere 2024.xx.y, aber auch die dazwischen liegenden Weekly Pakete. An sofort sind nur noch bestimmte Features unterschiedlich/lizenzpflichtig. Die Version ist aber immer gleich.

**Reply by ignoreigor** [reply]
Danke... Ja irgendwie so habe ich mir das gedacht. Wobei ich es nach dem erneuten Hören des Podcasts (in 1,5 facher Geschwindigkeit) auch nicht rausgehört habe und mich frage, ob der Plan aufgeht. Da bleibt doch nur zu hoffen, dass niemand falsch abbiegt!

**Reply by lab-at-nohl** [reply]
Oh ja, das habe ich mir auch gedacht. Der Podcast ist interessant, aber dafür sicherlich nicht hinreichend. 

Ich persönlich hoffe aber auf eine rege Community auch und gerade mit "kleinen" Installationen. Grundsätzlich kommt es mir sehr entgegen, nicht auf ein Umarbeiten der aktuellen Entwicklungen in einen Community Pfad warten zu müssen.

**Reply by My1** [reply]
> Der Entwicklungspfad beinhaltet insbesondere 2024.xx.y, aber auch die dazwischen liegenden Weekly Pakete. An sofort sind nur noch bestimmte Features unterschiedlich/lizenzpflichtig. Die Version ist aber immer gleich.

gibt es dazu genauer details was lizenzpflichtig ist, vor allem auch eine Interessante frage, wie Nutzungsbeschränkungen und open source miteinander interagieren, da man ja rein künstliche Limits wie eine Begrenzung der Zahl der Nutzer einfach rauspatchen könnte.

**Reply by lab-at-nohl** [reply]
Ich nehme die Info, was lizenzpflichtig ist, aus dem Quellcode. Siehe die Liste... Weitere Details habe ich nicht, außer, dass ein Patch mit dem nächstem Update wieder überschrieben würde. 

Nach meiner Kenntnis steht *alles* unter open source Lizenzen, frei aber nicht im Sinne von Freibier. Da ich nicht zum Hersteller gehöre, erlaube ich mir dazu zu sagen, @My1, there are infinitely many ways to get to the same random result. ;-)

**Reply by JPRuehmann** [reply]
Ich habe die Infos im Log File gefunden dachdem ich dieses aktiviert habe sonst kein Hinweis, nirgends. Worauf ebenfals nirgends Infos zu finden sind ist die Beschreibung des Zwecks der einzelnen Module.

Welche Module unter welcher Lizenz veröffentlicht werden ist Sache des Unternehmens, Die Nutzer (u.U. auch Kunden) Dumm sterben zu lassen ist aber Suboptimal und zwar auch aus Marketing Sicht.

**Reply by selfarian** [reply]
Hi,

ich weiß nicht, ob ich mich ein bisschen doof anstelle, aber kann es sein, dass es keinen "freien" Download von Tine mehr gibt? Also nur noch als Source-Code, nicht mehr als Package? Oder hab ich nur Tomaten auf den Augen?

Vielen Dank schon einmal :)

**Reply by My1** [reply]
\<5 user sollte hier gehen
https://github.com/tine-groupware/tine/releases/tag/2023.11.13

**Reply by selfarian** [reply]
@My1 Danke, da war ich auch schon unterwegs, aber da gibt es doch nur unten die Packages, die aber auch nur Source Code sind, oder sehe ich das falsch? Irgendwie stehe ich im Wald und sehe die Bäume nicht...

**Reply by My1** [reply]
hoffe das sollte deutlich genug sein.
![image](https://github.com/user-attachments/assets/7ac617a3-814b-42ec-824d-e24ac2a82c58)

**Reply by selfarian** [reply]
Schwer zu sehen, da sind lauter Pfeile im Weg ;)
Komisch, ich dachte ich hatte schon alle drei ausprobiert... Naja, jetzt geht es auf jeden Fall, Danke!

Kannst Du mir zufällig sagen: Zählt der Admin-User zu den fünf Usern dazu? Muss mir da dann ggf. etwas überlegen, sobald meine Jungs groß genug sind für eigene E-Mails wird es dann etwas knapp :)

**Reply by lab-at-nohl** [reply]
> Kannst Du mir zufällig sagen: Zählt der Admin-User zu den fünf Usern dazu? Muss mir da dann ggf. etwas überlegen, sobald meine Jungs groß genug sind für eigene E-Mails wird es dann etwas knapp :)

Siehe in meinem Ausgangspost die Nr. 3. Ich weiß nicht, welche Funktionalität du nutzt. Wahrscheinlich besteht aber gar keine Beschränkung der Anzahl...

**Reply by selfarian** [reply]
> > Kannst Du mir zufällig sagen: Zählt der Admin-User zu den fünf Usern dazu? Muss mir da dann ggf. etwas überlegen, sobald meine Jungs groß genug sind für eigene E-Mails wird es dann etwas knapp :)
> 
> Siehe in meinem Ausgangspost die Nr. 3. Ich weiß nicht, welche Funktionalität du nutzt. Wahrscheinlich besteht aber gar keine Beschränkung der Anzahl...

Ok, danke, dann bin ich erleichtert :) mich hatte das gerade etwas überrascht. Wir nutzen nur die Grundfunktionalitäten, also Mails, Kalender, Kontakte und eben dann active sync zum synchronisieren mit dem Handy. Das taucht glücklicherweise in deiner Aufzählung nicht auf :)

**Comment by selfarian** [comment]
Hi,

Danke für den Hinweis! Du hast mir glaube einen Haufen Ärger erspart :) Jetzt muss ich wohl doch mal auf PHP8 gehen :)

**Comment by JPRuehmann** [comment]
Ein kleiner Hinweis, die .htaccess sollte ebenfalls beim Update wiederverwendet werden.
Ohne findet Tine nicht alles, z.B. das Logo auf der Anmeldeseite.

Was mich stört ist das die nicht installierbaren komponenten in der Setup.php keinerlei Hinweis, weder warum noch das sie nicht installiert werden können. Die seite wird einfach kommentarlos ohne änderung neu geladen. Was los ist findet mann nur raus wenn mann die Logs aktiviert. Dort steht dann das die Lizenz fehlt.

Ich habe auch nirgends Hinweise gefunden was diese Komponenten spezielles machen, weshalb z.B. das Handbuch eine Lizenz benötigt sales aber nicht.

**Reply by lab-at-nohl** [reply]
Setup.php ist eh nicht das Gelbe vom Ei. 😁

Für die lizenzpflichtigen Features hatte ich oben mal eine aktuelle Liste gepostet... Kann natürlich den Hinweis im Programm nicht ersetzen. Warum z.B. das Handbuch Geld kostet, kann ich nur erraten. Ich würde aber vermuten, dass es kostenlos werden würde, wenn die Arbeit daran im Wesentlichen durch die Community käme; so muss halt das Personal dafür bezahlt werden.

**Comment by JPRuehmann** [comment]
Da wäre sale aber logischer. Wer das Modul benötigt verdient wahrscheinlich auch Geld damit während das Handbuch jeder benötigt. Wenn es das ist was ich denke. Könnte ja auch wie das FAQ Modul zum selbst erstellen von Handbüchern gedacht sein.

Was EFile, GDPR und DFCom sind/machen weiss ich dennoch nicht.

**Reply by pschuele** [reply]
hier ein paar infos, die ich  zu den genannten tine apps gefunden habe :

### GDPR (DSGVO)

am besten den podcast hören: https://www.tine-groupware.de/podcast/10-die-dsgvo-anwendung/ :wink: 

> Die Anwendung DSGVO ist im eigentlichen Sinne keine eigene Anwendung, sondern ein modularer Teil des Adressbuchs sowie der Stammdaten und gibt so die Möglichkeit, die Herkunft und den Verwendungszweck von Daten zu erfassen und eine entsprechende Bearbeitungshistorie zu führen. Dies ermöglicht eine DSGVO-konforme Erfassung personenbezogener Daten.

### DFCom

> DFCom steht für DataFox Communicator und ist genau das: ein Kommunikationstool zwischen tine und den DataFox-Terminals. Hier geht es in erster Linie, aber nicht ausschließlich, um die Arbeitszeiterfassungs-Terminals von DataFox. Das DFCom-Modul bietet auch die Option, andere DataFox-Endgeräte anzubinden.

### EFile (EAkte)

> Mit fortschreitender Digitalisierung inklusive deren gesetzlichen Rahmenbedingungen ist die Einführung einer elektronischen Aktenführung zunehmend erforderlich.
Dies hat uns veranlasst, in enger Zusammenarbeit mit Kunden und in Anlehnung an die rechtlichen, fachlichen sowie funktionalen Anforderungen zum „Baustein E-Akte“ des „Organisationskonzepts elektronische Verwaltungsarbeit“ vom Bundesministerium des Innern (BMI), die Anwendung eAkte zu entwickeln.
Die elektronische Abbildung entspricht dabei den Grundsätzen ordnungsgemäßer Aktenführung (gesetzlich nicht normiert) und der verwaltungs- bzw. behördlichen Schriftgutsammlung.
Als E-Akte wird allgemein eine digitale Datensammlung bezeichnet, die nach dem Vorbild herkömmlicher Akten (auf Papier) aufgebaut ist.

**Comment by JPRuehmann** [comment]
Danke, Da ich Tine ja rein Privat nutze brauche ich die Optionen ja nicht, wie übrigens auch Sale.
Was vielleicht für den privaten Einsatz interessant sein könnte sind Die Anleitung und FAQ.
Bin gespannt wie es weiter geht.
Was noch interessant währe sind 0Auth und vcf Import.
